src/games/strategy/triplea/ui/MapData.java

```diff
@@ -700,6 +701,50 @@ private Rectangle getBoundingRectWithTranslate(final List<Polygon> polys, final
   }
 
   /**
+   * Find the best rectangle inside the territory to place the name in. Finds the rectangle
+   * that can fit the name, that is the closest to the vertical center, and has a large width at
+   * that location. If there isn't any rectangles that can fit the name then default back to the
+   * bounding rectangle.
+   */
+  public Rectangle getBestTerritoryNameRect(final Territory t, final FontMetrics fm) {
+    final Rectangle territoryBounds = getBoundingRect(t);
+    Rectangle maxRectangle = territoryBounds;
+    final int minX = territoryBounds.x;
+    final int minY = territoryBounds.y;
+    final int maxX = minX + territoryBounds.width;
+    final int maxY = minY + territoryBounds.height;
+    final int centerY = minY + territoryBounds.height / 2;
+    final List<Polygon> polygons = m_polys.get(t.getName());
+    final int minWidth = fm.stringWidth(t.getName());
+    final int increment = fm.getAscent();
+    int maxScore = 0;
+    for (int startX = minX; startX < maxX - increment; startX += increment) {
+      for (int startY = minY; startY < maxY - increment; startY += increment) {
+        for (int endX = maxX; endX > startX; endX -= increment) {
+          final Rectangle current = new Rectangle(startX, startY, endX - startX, increment);
+          final int verticalFromEdge = territoryBounds.height / 2 - Math.abs(centerY - (startY + increment));
+          final int score = verticalFromEdge * current.width;
+          if (current.width > minWidth && score > maxScore) {
+            boolean isContained = false;
+            for (final Polygon polygon : polygons) {
+              if (polygon.contains(current)) {
```

Comment:
Not sure exactly what your point or question is here.

verticalFromEdge will start at its min value then moves to max at vertical center then move back to min value (ex. 1 2 3 2 1) and yes is only influenced by Y.

width is influenced by both left X and right X (1st and 3rd loops).

The idea is to find the rectangle that is the widest and closest to vertical center.

A 'break' is used in the inner most loop since it loops backwards starting with the widest rectangle at the given point (X, Y) so as soon as you get a success then you don't need to check any other rectangles at that point since they will all have a lower score.